src/lib/components/SgtDialog/SgtDialog.vue

```diff
@@ -0,0 +1,156 @@
+            <v-icon
+              class="close-btn"
+              :color="invertedColor"
+              @click="$close('close')"
+              >mdi-close</v-icon
+            >
+          </div>
+        </v-card-title>
+        <v-divider></v-divider>
+        <v-card-text class="content-container">
+          <div v-if="modalContentType === 'text'" class="text-content">
+            <p>{{ modalContent }}</p>
+          </div>
+          <template v-else>
+            <div class="text-content" v-html="modalContent"></div>
+          </template>
+        </v-card-text>
+        <v-card-actions class="button-container">
+          <v-btn
+            :color="dialogColor"
+            @click="$close('ok')"
+            dark
+            v-if="modalType !== 'prompt'"
+            >{{ okButtonLabel }}</v-btn
+          >
+          <template v-else>
+            <v-btn :color="dialogColor" @click="$close('yes')" dark>{{
+              yesButtonLabel
+            }}</v-btn>
+            <v-btn :color="dialogColor" @click="$close('no')" dark outlined>{{
+              noButtonLabel
+            }}</v-btn>
+          </template>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, PropSync } from "vue-property-decorator";
+import { DialogComponent } from "vue-modal-dialogs";
+import { SgtDialogConst } from "./SgtDialog.constant";
+
+@Component({
+  name: "SgtDialog",
+  components: {},
+})
+export default class SgtDialog extends DialogComponent<boolean> {
+  @PropSync("showModal", { required: false, default: false })
+  private dialog: boolean;
+  @Prop({ required: true, default: "message" }) private modalType:
+    | "prompt"
+    | "message";
+  @Prop({ required: false }) private modalTitle: string;
+  @Prop({ required: true }) private modalContent: string;
+  @Prop({ required: false, default: "neutral" }) private infoType:
+    | "success"
+    | "alert"
+    | "warning"
+    | "neutral";
+  @Prop({ required: false, default: "text" }) private modalContentType:
+    | "text"
+    | "html";
+  @Prop({ required: false, default: SgtDialogConst.modalWidth })
+  private modalWidth: string;
+  @Prop({ required: false, default: SgtDialogConst.okButtonLabel })
+  private okButtonLabel: string;
+  @Prop({ required: false, default: SgtDialogConst.yesButtonLabel })
+  private yesButtonLabel: string;
+  @Prop({ required: false, default: SgtDialogConst.noButtonLabel })
+  private noButtonLabel: string;
+  @Prop({ required: false, default: true }) private persistent: boolean;
+
+  private dialogConst = SgtDialogConst;
+
+  get dialogColor() {
+    return this.dialogConst.infoTypeColors[this.infoType];
+  }
+  get invertedColor() {
+    return this.infoType === "neutral"
+      ? this.dialogConst.infoTypeColors.neutral
+      : "#FFFFFF";
+  }
+
+  get getModalTitle() {
+    return this.modalTitle ? this.modalTitle : this.infoType;
```

Comment:
'neutral' info type won't look good as the title. Instead, you could have a default value for modalTitle like "Action required" or something generic.

Reply:
done